pylistenbrainz/playlist.py

```diff
@@ -0,0 +1,268 @@
+    :type public: bool
+
+    """
+    def __init__(self,
+                 identifier,
+                 annotation=None,
+                 creator=None,
+                 date=None,
+                 title=None,
+                 algorithm_metadata=None,
+                 collaborators=None,
+                 last_modified_at=None,
+                 public=True):
+        self.identifier = identifier
+        self.annotation = annotation
+        self.creator = creator
+        self.date = date
+        self.title = title
+        self.algorithm_metadata = algorithm_metadata
+        self.collaborators = collaborators or []
+        self.last_modified_at = last_modified_at
+        self.public = public
+
+        self.url = _mbid_to_url(identifier, 'playlist')
+
+    def __repr__(self):
+        return('<pylistenbrainz.PlaylistMetadata url="{}">'.format(self.url))
+
+    def identifier(self):
+        """The MBID of the playlist."""
+        return self.identifier
+
+    def url(self):
+        """A URL where the playlist can be found."""
+        return self.url
+
+
+def _maybe_add(data, key, value):
+    if key:
+        data[key] = value
+
+
+class PlaylistTrack():
+    """A single item in a Listenbrainz playlist.
+
+    :param identifier: the MBID of the recording
+    :type identifier: str
+    :param creator: the username of the playlist's creator
+    :type creator: str
+    :param title: the title of the playlist
+    :type title: str
+    :param added_at: the timestamp when the track was added to the playlist
+    :type added_at: datetime.datetime, optional
+    :param added_by: the username of the user who added this track
+    :type added_by: str, optional
+    :param artist_identifiers: the MBID of the recording artist(s)
+    :type artist_identifiers: List[str], optional
+    """
+    def __init__(self,
+                 identifier,
+                 creator,
+                 title,
+                 added_at=None,
+                 added_by=None,
+                 artist_identifiers=None):
+        self.identifier = identifier
+        self.creator = creator
+        self.title = title
+        self.added_at = added_at
+        self.added_by = added_by
+        self.artist_identifiers = artist_identifiers
+
+        self.url = _mbid_to_url(identifier, 'recording')
+
+    def __repr__(self):
+        return('<pylistenbrainz.PlaylistTrack url="{}">'.format(self.url))
+
+    def identifier(self):
+        """The MBID of the recording."""
+        return self.identifier
+
+    def url(self):
+        """A URL where the recording can be found."""
+        return self.url
+
+    def to_jspf(self):
+        """Return the playlist entry as a fragment of JSPF data."""
+        ext = {}
+        if self.added_at:
+            ext['added_at'] = self.added_at.isoformat()
+        _maybe_add(ext, 'added_by', self.added_by)
+        _maybe_add(ext, 'artist_identifier', self.artist_identifiers)
```

Comment:
In [LB-1058](https://tickets.metabrainz.org/browse/LB-1058) we noticed some inconsistencies in our specification of this field and decided to standardise on `artist_identifiers`

Reply:
Thanks for the comment! Have updated the field name in the local API, and the code will handle both options from the remote API for now.